lib/crawler.js

```diff
@@ -297,39 +465,34 @@ Crawler.prototype.start = function() {
     });
 
     crawler.running = true;
+
+    /**
+     * Fired when the crawl starts. This event gives you the opportunity to
+     * adjust the crawler's configuration, since the crawl won't actually start
+     * until the next processor tick.
+     * @event Crawler#fetchstart
+     */
     crawler.emit("crawlstart");
 
     return crawler;
 };
 
-/*
-    Public: Fetch condition that looks at robots.txt rules. Method is replaced
-    once the content of robots.txt has been loaded.
-
-    queueItem - A queueItem or a URL string
-
-    Returns a boolean
+/**
+ * Determines whether robots.txt rules allows the fetching of a particular URL
+ * or not
+ * @param  {String} url The full URL of the resource that is to be fetched (or not)
+ * @return {Boolean}    Returns true if the URL is allowed to be fetched, otherwise false
  */
-Crawler.prototype.urlIsAllowed = function (queueItem) {
+Crawler.prototype.urlIsAllowed = function(url) {
```

Comment:
I think this was my attempt to make this method more flexible at some point. Coming back to it now, though, I just found it to make the method more confusing. And when calling this method, you're always going to have access to the queue item's URL if you have access to the queue item itself